### PR TITLE
Update mathlib4 to v4.11.0-rc2

### PR DESCRIPTION
**Pull request by Parcly-Taxel**
Includes a splitting of the 1500+-line `Carleson.DiscreteCarleson` into three more manageable files.

### CODE REVIEW

File: Carleson/ToMathlib/Height.lean [hunk 1]
```diff
@@ -367,20 +339,17 @@ lemma height_eq_coe_iff (x : α) (n : ℕ) :
...
--- This is from mathlib, but has too strict requirements, PartialOrder is too strong, Preorder suffices
-theorem mem_minimals_iff_forall_lt_not_mem'' {x : α} {s : Set α} :
-    x ∈ minimals (· ≤ ·) s ↔ x ∈ s ∧ ∀ ⦃y⦄, y < x → y ∉ s :=
-  mem_minimals_iff_forall_lt_not_mem' (· < ·)
-
-lemma mem_minimal_univ_iff_height_eq_zero (a : α) :
-    a ∈ minimals (·≤·) Set.univ ↔ height a = 0 := by
-  simp [mem_minimals_iff_forall_lt_not_mem'', height_eq_zero_iff]
+lemma minimal_iff_height_eq_zero (a : α) : Minimal Set.univ a ↔ height a = 0 := by
+  simp [minimal_iff_forall_lt_not_mem, height_eq_zero_iff]
```
Comment (nomeata) [inline Carleson/ToMathlib/Height.lean:347]: `Set.univ` used as a predicate typechecks, but has really bad simp behavior, it seems. Maybe better like this:
```suggestion
lemma minimal_iff_height_eq_zero (a : α) :
    Minimal (fun _ => True) a ↔ height a = 0 := by
  simp [minimal_iff_forall_lt, height_eq_zero_iff]
```

File: Carleson/ToMathlib/Height.lean [hunk 2]
```diff
@@ -367,20 +339,17 @@ lemma height_eq_coe_iff (x : α) (n : ℕ) :
...
 
 lemma mem_minimal_le_height_iff_height (a : α) (n : ℕ) :
-    a ∈ minimals (·≤·) { y | n ≤ height y } ↔ height a = n := by
+    Minimal { y | n ≤ height y } a ↔ height a = n := by
```
Comment (nomeata) [inline Carleson/ToMathlib/Height.lean:350]: Likewise
```
lemma mem_minimals_le_height_iff_height (a : α) (n : ℕ) :
    Minimal (fun y => n ≤ height y) a ↔ height a = n
```

File: Carleson/DiscreteCarleson.lean [hunk 3]
```diff
@@ -26,7 +26,7 @@ def aux𝔐 (k n : ℕ) : Set (𝔓 X) :=
   {p ∈ TilesAt k | 2 ^ (-n : ℤ) * volume (𝓘 p : Set X) < volume (E₁ p) }
 
 /-- The definition `𝔐(k, n)` given in (5.1.4) and (5.1.5). -/
-def 𝔐 (k n : ℕ) : Set (𝔓 X) := maximals (·≤·) (aux𝔐 k n)
+def 𝔐 (k n : ℕ) : Set (𝔓 X) := {m | Maximal (aux𝔐 k n) m}
```
Comment (fpvandoorn) [inline Carleson/DiscreteCarleson.lean:29]: Here and elsewhere: this should be `Maximal (· ∈ aux𝔐 k n) m`

**Comment by fpvandoorn** [comment]
I think I'll wait to see if the panics here are fixed quickly; see [this Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/PANIC.20at.20Lean.2EExpr.2EappArg!.20Lean.2EExpr.3A896.3A15.3A.20application.20exp)

**Comment by Parcly-Taxel** [comment]
> I think I'll wait to see if the panics here are fixed quickly; see [this Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/PANIC.20at.20Lean.2EExpr.2EappArg!.20Lean.2EExpr.3A896.3A15.3A.20application.20exp)

The link doesn't work!

**Comment by Parcly-Taxel** [comment]
@fpvandoorn how about you just merge this PR? The panic is not fatal to the build.

**Comment by fpvandoorn** [comment]
This PR is really hard to review because you're doing so much things at the same time, and there are a few things I have questions about. Please don't put everything in the same PR the next time.

**Comment by fpvandoorn** [comment]
I guess there were some major changes that made required some big changes, thanks for doing that.

**Comment by fpvandoorn** [comment]
I'll merge it as-is, and clean up afterwards.

**Pull request merged**